app/src/main/java/com/kickstarter/ui/activities/ThreadActivity.kt

```diff
@@ -26,28 +26,40 @@ class ThreadActivity :
     private lateinit var ksString: KSString
 
     private val adapter = RepliesAdapter(this)
+    private val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, true)
+    private lateinit var recyclerViewPaginator: RecyclerViewPaginator
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityThreadLayoutBinding.inflate(layoutInflater)
         setContentView(binding.root)
         ksString = environment().ksString()
-
+        recyclerViewPaginator = RecyclerViewPaginator(binding.commentRepliesRecyclerView, { viewModel.inputs.nextPage() }, viewModel.outputs.isFetchingReplies(), false)
+        linearLayoutManager.stackFromEnd = true
         binding.commentRepliesRecyclerView.adapter = adapter
+        binding.commentRepliesRecyclerView.layoutManager = linearLayoutManager
 
         this.viewModel.getRootComment()
             .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
             .subscribe { comment ->
-                configureRootCommentView(comment)
+                adapter.updateRootCommentCell(comment)
             }
 
-        this.viewModel.onCommentReplies()
+        this.viewModel
+            .onCommentReplies()
             .compose(bindToLifecycle())
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe { comments ->
-                this.adapter.takeData(comments)
+            .subscribe {
+                this.adapter.takeData(it.first.reversed(), it.second)
+            }
+
+        viewModel.outputs.isFetchingReplies()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                // binding.commentsLoadingIndicator.isVisible = it
```

Comment:
Nothing is done in this subscription, what's it's purpose?

Reply:
fixed add loading Indicator